src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlStringTranslator.cs

```diff
@@ -0,0 +1,431 @@
+    /// </remarks>
+    public class NpgsqlStringTranslator : IMethodCallTranslator
+    {
+        #region MethodInfo
+
+        [NotNull] static readonly MethodInfo Concat = typeof(string).GetRuntimeMethod(nameof(string.Concat), new[] { typeof(string), typeof(string) });
+        [NotNull] static readonly MethodInfo Contains = typeof(string).GetRuntimeMethod(nameof(string.Contains), new[] { typeof(string) });
+        [NotNull] static readonly MethodInfo EndsWith = typeof(string).GetRuntimeMethod(nameof(string.EndsWith), new[] { typeof(string) });
+        [NotNull] static readonly MethodInfo StartsWith = typeof(string).GetRuntimeMethod(nameof(string.StartsWith), new[] { typeof(string) });
+        [NotNull] static readonly MethodInfo IndexOfString = typeof(string).GetRuntimeMethod(nameof(string.IndexOf), new[] { typeof(string) });
+        [NotNull] static readonly MethodInfo IndexOfChar = typeof(string).GetRuntimeMethod(nameof(string.IndexOf), new[] { typeof(char) });
+        [NotNull] static readonly MethodInfo IsNullOrWhiteSpace = typeof(string).GetRuntimeMethod(nameof(string.IsNullOrWhiteSpace), new[] { typeof(string) });
+        [NotNull] static readonly MethodInfo Trim = typeof(string).GetRuntimeMethod(nameof(string.Trim), Type.EmptyTypes);
+        [NotNull] static readonly MethodInfo TrimWithChars = typeof(string).GetRuntimeMethod(nameof(string.Trim), new[] { typeof(char[]) });
+        [NotNull] static readonly MethodInfo TrimEndWithChars = typeof(string).GetRuntimeMethod(nameof(string.TrimEnd), new[] { typeof(char[]) });
+        [NotNull] static readonly MethodInfo TrimStartWithChars = typeof(string).GetRuntimeMethod(nameof(string.TrimStart), new[] { typeof(char[]) });
+        [NotNull] static readonly MethodInfo Substring = typeof(string).GetTypeInfo().GetDeclaredMethods(nameof(string.Substring)).Single(m => m.GetParameters().Length == 2);
+        [NotNull] static readonly MethodInfo Replace = typeof(string).GetRuntimeMethod(nameof(string.Replace), new[] { typeof(string), typeof(string) });
+
+        // The following exist as optimizations in netcoreapp20
+        [NotNull] static readonly MethodInfo TrimWithSingleChar = typeof(string).GetRuntimeMethod(nameof(string.Trim), new[] { typeof(char) });
+        [NotNull] static readonly MethodInfo TrimEndNoParam = typeof(string).GetRuntimeMethod(nameof(string.TrimEnd), new Type[0]);
+        [NotNull] static readonly MethodInfo TrimEndSingleChar = typeof(string).GetRuntimeMethod(nameof(string.TrimEnd), new[] { typeof(char) });
+        [NotNull] static readonly MethodInfo TrimStartNoParam = typeof(string).GetRuntimeMethod(nameof(string.TrimStart), new Type[0]);
+        [NotNull] static readonly MethodInfo TrimStartSingleChar = typeof(string).GetRuntimeMethod(nameof(string.TrimStart), new[] { typeof(char) });
+
+        #endregion
+
+        /// <summary>
+        /// The backend version to target.
+        /// </summary>
+        [CanBeNull] readonly Version _postgresVersion;
```

Comment:
None of the current translations seem to be version-specific, did you include this in case we need to be in the future? If so I'd prefer to not have it (we can easily add it later as required).

Reply:
Kind of... This originated from the copy/paste of another translator, then I kept for a potential compatibility check on the existing translators. But there's not really a reason for it at this stage (plus it would expand the scope of this PR). 

I'll omit before merging.